common/txmgr/types/attempt_builder.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services"
+)
+
+// AttemptBuilder takes the base unsigned transaction + optional parameters (tx type, gas parameters)
+// and returns a signed TxAttempt
+// it is able to estimate fees and sign transactions
+// H - chain head type
+// F - chain fee type
+// A - chain address type
+// HA - chain hash type
+// GU - chain gas unit type
+// T - tx type (will be replaced in future)
+// TA - tx attempt type (will be replaced  in future)
+//
+//go:generate mockery --quiet --name AttemptBuilder --output ./mocks/ --case=underscore
+type AttemptBuilder[H Head, F Fee, A any, HA any, GU any, T any, TA any] interface {
```

Comment:
Would this format make it easier to doc the type constraints?
```suggestion
type AttemptBuilder[
	// H is chain head type
	H Head,
	F Fee, // F is chain fee type
	A any, 
	HA any, 
	GU any, 
	T any, 
	TA any,
] interface {
```